staging/src/k8s.io/kubectl/pkg/cmd/get/get.go

```diff
@@ -619,6 +620,9 @@ func (o *GetOptions) watch(f cmdutil.Factory, args []string) error {
 		TransformRequests(o.transformRequests).
 		Do()
 	if err := r.Err(); err != nil {
+		if errors.Is(err, resource.ErrMultipleResourceTypes) {
+			return i18n.Errorf("watch is only supported on individual resources and resource collections - more than 1 resource was found")
+		}
```

Comment:
The proposed error is confusing, I'd just leave the one from `ErrMultipleResourceTypes` which will result in:
```
$ kubectl get po,no --watch
error: you may only specify a single resource type
```

Reply:
This effectively means this PR is only about removing the extra `if multipleGVKsRequested(infos)` check because it's actually never true and not changing any logic. Do we need a custom error in that case?

Reply:
Think about it this way, what important information are you trying to convey to the user with that check? As I'm reading it, there's none.

Reply:
I think that the idea is that the error message is not super clear, because the issue is not with `kubectl get`, but with `--watch`. So when you see `you may only specify a single resource type`, you don't immediately see what's wrong.

Reply:
I didn't really propose the message, I just copied it from the if condition down there, but in any case, I amended the PR to remove the error message and it now only prints `you may only specify a single resource type`. So feel free to review and approve 😉 Thanks.